### DISCUSSION TITLE
[Q&A] Agent update automatically via VBS

### DISCUSSION THREAD

**Discussion by timpoldo** [glpi-project/glpi-agent#624] [Q&A]
### Documentation & bug reporting acknowledgment

Yes, I read it

### Describe your problem

I use a VBS script in my domain to deploy and install agent on all  windows machine, and al works great.
So far, the clients downloaded v.1.4.
Now, i've update it VBS script (the source is [here](https://raw.github.com/glpi-project/glpi-agent/develop/contrib/windows/glpi-agent-deployment.vbs) ) and in this way only the NEW clients download the v.1.7.1 agent and install it.

_My question_: how can i force the update to v.1.7.1 for the "old" installed clients?
The existing clients doesn't auto-update themself?

Thanks in advance.

**Comment by g-bougard** [comment]
Hi @timpoldo 

there's no auto-update support actually.

You still have to update that vbs script to include your own configuration (essentially the `SERVER` in `SetupOptions` variable).

When you want to update the agent, you have to update the version `SetupVersion` with the desired version. You should know most people using this script doesn't permit computer to access internet directly so they have to provide the msi in a shared and trusted place. In that such case, auto-update is just not possible.

The auto-update may be implemented in the future but it will have to cover as much possible update scenario as possible.

**Reply by MrDick-11** [reply]
but when I'm trying to update with vbs script 1.5 to 1.8, it just installs a new copy of glpi agent. So I have now 1.5 and 1.8
How to delete 1.5 while installing 1.8?

**Reply by g-bougard** [reply]
Hi @MrDick-11 

as far I know, this is only possible if 1.5 version was 32 bits version. In that case, the 1.8 is not an upgrade but the installation of 64 bits version.

To uninstall, you have to run the 1.5 32 bits installer with [msiexec.exe](https://learn.microsoft.com/en-us/windows/win32/msi/standard-installer-command-line-options) uninstall option. You should also find a more exact commandline sample you can run in registry into the `UninstallString` key of the installtion related registry key under `HKEY_LOCAL_MACHINE\SOFTWARE\WOW6432Node\Microsoft\Windows\CurrentVersion\Uninstall`.

**Comment by jeremiassamuelzitnik** [comment]
I noticed that VBS is being deprecated. I've prepared a script and executable in PowerShell to replace it. What are your thoughts on this?

Also, here's the GitHub link for reference:

https://github.com/jeremiassamuelzitnik/GLPI/releases

**Reply by g-bougard** [reply]
You can open a PR for integration in glpi-agent `contrib/windows` folder and I'll dive deeper in it. You still can probably keep a separated arch var, even if only x64 is actually supported. We no more produces x86 arch, but Arm64 may be in a more or less near future. Of course, we can also modify the script later for that.

**Comment by apohutenumera** [comment]
Hi, is there any update on this ?
regards
